Source/Dafny/Compilers/Compiler.cs

```diff
@@ -1784,8 +1795,18 @@ protected class NullClassWriter : IClassWriter {
           }
           v.Visit(f);
         } else if (member is Method m) {
-          if (m.Body == null && !(c is TraitDecl && !m.IsStatic) &&
-              !(!DafnyOptions.O.DisallowExterns && (Attributes.Contains(m.Attributes, "dllimport") || (IncludeExternMembers && Attributes.Contains(m.Attributes, "extern"))))) {
+          if (Attributes.Contains(m.Attributes, "synthesize") &&
+              DafnyOptions.O.Synthesize) {
+            if (m.IsStatic && m.Outs.Count > 0 && m.Body == null) {
+              classWriter.SynthesizeMethod(m, CombineAllTypeArguments(m), true, true, false);
+            } else {
+              Error(m.tok, "Method {0} is annotated with :synthesize but " +
+                           "does not have the correct specification or has a " +
```

Comment:
Can you be more specific about what "the correct specification" is (i.e. static and returning at least one value) in the error message?

Reply:
Changed to "...is not static, has a body, or does not return anything"